pkg/clients/cloudsql/cloudsql.go

```diff
@@ -283,10 +296,17 @@ func LateInitializeSpec(spec *v1beta1.CloudSQLInstanceParameters, in sqladmin.Da
 
 // IsUpToDate checks whether current state is up-to-date compared to the given
 // set of parameters.
-func IsUpToDate(in *v1beta1.CloudSQLInstanceParameters, currentState sqladmin.DatabaseInstance) bool {
-	currentParams := &v1beta1.CloudSQLInstanceParameters{}
-	LateInitializeSpec(currentParams, currentState)
-	return cmp.Equal(in, currentParams, cmpopts.IgnoreInterfaces(struct{ resource.AttributeReferencer }{}))
+func IsUpToDate(name string, in *v1beta1.CloudSQLInstanceParameters, currentState *sqladmin.DatabaseInstance) (bool, error) {
+	genState, err := copystructure.Copy(currentState)
+	if err != nil {
+		return false, err
+	}
+	_, ok := genState.(*sqladmin.DatabaseInstance)
+	if !ok {
+		return false, errors.New("unable to type case deepcopy to *sqladmin.DatabaseInstance")
+	}
+	GenerateDatabaseInstance(name, *in, genState.(*sqladmin.DatabaseInstance))
+	return cmp.Equal(genState, currentState, cmpopts.IgnoreFields(sqladmin.DatabaseInstance{}, "Settings.IpConfiguration.ForceSendFields")), nil
```

Comment:
Naming nitpick: `observed` and `desired` might be more idiomatic (to Kubernetes) than `currentState` and `genState`.